api/v1beta1/azuremanagedcontrolplane_webhook.go

```diff
@@ -676,3 +683,26 @@ func (m *AzureManagedControlPlane) validateIntegerStringGreaterThanZero(input *s
 
 	return allErrs
 }
+
+// validateIdentityProfile validates an AutoScalerProfile.
+func (m *AzureManagedControlPlane) validateIdentityProfile(_ client.Client) error {
+	var allErrs field.ErrorList
+
+	if m.Spec.IdentityProfile != nil {
+		if m.Spec.IdentityProfile.Type == ManagedControlPlaneIdentityTypeUserAssigned {
+			if m.Spec.IdentityProfile.UserAssignedIdentityResourceID == "" {
+				allErrs = append(allErrs, field.Invalid(field.NewPath("Spec", "IdentityProfile", "UserAssignedIdentityResourceID"), m.Spec.IdentityProfile.UserAssignedIdentityResourceID, "cannot be empty if IdentityProfile.Type is UserAssigned"))
+			}
+		} else {
+			if m.Spec.IdentityProfile.UserAssignedIdentityResourceID != "" {
+				allErrs = append(allErrs, field.Invalid(field.NewPath("Spec", "IdentityProfile", "UserAssignedIdentityResourceID"), m.Spec.IdentityProfile.UserAssignedIdentityResourceID, "should be empty if IdentityProfile.Type is SystemAssigned"))
+			}
+		}
+	}
+
+	if len(allErrs) > 0 {
+		return kerrors.NewAggregate(allErrs.ToAggregate().Errors())
+	}
+
+	return nil
```

Comment:
I agree with Jon that we could return `allErrs`, but is there a reason for aggregating the errors which I may be missing?

Reply:
If you mean "why do we return multiple errors" then in general, we want to return as much information as possible to the user for each request so they get one "spec.A is invalid, spec.B is invalid, ..." error on the first request vs. "spec.A is invalid." Fix spec.A. "spec.B is invalid." Fix spec.B...

If you mean "why are we calling `kerrors.NewAggregate` then that's because that's one way of turning a `[]error` into a single `error`.

Reply:
I think I'd have to refactor more to use the simpler approach, as the `Validate` method is doing the error wrapping by enumerating through a bunch of validator funcs (`validateIdentityProfile` is a new one I added here) and each of those funcs is returning an `err`.

Reply:
Ah I missed that, this is definitely fins as-is then IMO.